public/main.js

```diff
@@ -16,22 +16,22 @@ function createCard(data) {
   $qrCodeDiv.appendChild($qrCode)
   $qrCodeDiv.appendChild($name)
 
-  const $cardRow = document.getElementById('cardrow')
-  $cardRow.appendChild($qrCodeDiv)
+  return $qrCodeDiv
 }
 
 window.onload = () => {
   fetch('/qrcards')
     .then(res => res.json())
-      .then(data => {
-        data.forEach((data) => {
-          createCard(data)
-        })
-        if (data.length > 0) {
-          const $cardHeader = document.getElementById('cardheader')
-          $cardHeader.classList.remove('hidden')
-        }
+    .then(data => {
+      data.forEach((data) => {
+        const $cardRow = document.getElementById('cardrow')
```

Comment:
better! but no reason to re-query for this element. You can graph it right before `data.forEach`